Roles/Neutral/Yandere.cs

```diff
@@ -0,0 +1,216 @@
+                writer.Write(lawyerId);
+                writer.Write(targetId);
+                AmongUsClient.Instance.FinishRpcImmediately(writer);
+                break;
+            case "":
+                if (!AmongUsClient.Instance.AmHost) return;
+                writer = AmongUsClient.Instance.StartRpcImmediately(PlayerControl.LocalPlayer.NetId, (byte)CustomRPC.RemoveYandereTarget, SendOption.Reliable);
+                writer.Write(lawyerId);
+                AmongUsClient.Instance.FinishRpcImmediately(writer);
+                break;
+        }
+    }
+
+    public static void ReceiveRPC(MessageReader reader)
+    {
+        // 方法实现
+        byte playerId = reader.ReadByte();
+        bool add = reader.ReadBoolean();
+        if (add)
+            LocateArrow.Add(playerId, new Vector3(reader.ReadSingle(), reader.ReadSingle(), reader.ReadSingle()));
+        else
+            LocateArrow.RemoveAllTarget(playerId);
+        byte bountyId = reader.ReadByte();
+        byte targetId = reader.ReadByte();
+
+        Target[bountyId] = targetId;
+        TargetArrow.Add(bountyId, targetId);
+    }
+    public static void ChangeRoleByTarget(PlayerControl target)
+    {
+        target.RpcMurderPlayerV3(target);
+        Utils.NotifyRoles();
+    }
+    public static bool KnowRole(PlayerControl player, PlayerControl target)
+    {
+        if (!KnowTargetRole.GetBool()) return false;
+        return player.Is(CustomRoles.Yandere) && Target.TryGetValue(player.PlayerId, out var tar) && tar == target.PlayerId;
+    }
+
+    public static void OnFixedUpdate(PlayerControl yandere)
+    {
+        List<byte> deList = new();
+        if (!IsEnable || !GameStates.IsInTask) return;
+        foreach (var lv in Main.ForYandere)
+        {
+            var si = Utils.GetPlayerById(lv);
+            foreach (var pcs in Main.AllAlivePlayerControls)
+            {
+                var posi = si.transform.position;
+                var diss = Vector2.Distance(posi, pcs.transform.position);
+                if (diss > 0.3f) continue;
+                if (Main.ForYandere.Contains(pcs.PlayerId)) continue;
+                if (!Main.NeedKillYandere.Contains(pcs.PlayerId) && !pcs.Is(CustomRoles.Yandere))
+                {
+                    Main.NeedKillYandere.Add(pcs.PlayerId);
+                    foreach (var yanderes in Main.AllAlivePlayerControls)
+                    {
+                        if (yanderes.Is(CustomRoles.Yandere))
+                        {
+                            var playerId = yanderes.PlayerId;
+                            NameColorManager.Add(yanderes.PlayerId,pcs.PlayerId, "#FF0000");
+                            var targetId = pcs.PlayerId;
+                            Target[playerId] = targetId;
+                            TargetArrow.Add(playerId, targetId);
+                        }
+                    }                                      
```

Comment:
Perfect example of how to write laggy code. 3 foreach cycles within each other put in FixedUpdatePatch. Is there really not a better way to do this?
If not, at least convert them into for cycles using the .ElementAt(i) method if you want better performance

Reply:
Damn, I didn't even notice that there are 3 foreach loops💀

Reply:
💀